src/libraries/System.Private.CoreLib/src/System/Buffers/Text/Base64Decoder.cs

```diff
@@ -68,7 +68,18 @@ private static unsafe OperationStatus DecodeFromUtf8(ReadOnlySpan<byte> utf8, Sp
 
                 if (maxSrcLength >= 24)
                 {
-                    byte* end = srcMax - 45;
+                    byte* end = srcMax - 88;
+                    if (Vector512.IsHardwareAccelerated && Avx512BW.IsSupported && (end >= src))
```

Comment:
Afair, `Avx512BW` implies `Vector512.IsHardwareAccelerated` or the latter is needed because the analyzer will complain?

Reply:
Vector512.IsHardwareAccelerated is there to weed out the scenario where throttling issues are there. Vector512.IsHardwareAccelerated returns false even if AVX512BW is supported if it has throttling issues like Cascade Lakes

Reply:
ah, good point